jmclient/jmclient/configure.py

```diff
@@ -529,13 +586,36 @@ def get_irc_mchannels():
             server_data["socks5_host"] = jm_single().config.get(section, "socks5_host")
             server_data["socks5_port"] = jm_single().config.get(section, "socks5_port")
 
-        for option, otype in req_fields:
+        for option, otype in irc_fields:
             val = jm_single().config.get(section, option)
             server_data[option] = otype(val)
         server_data['btcnet'] = get_network()
         configs.append(server_data)
-    return configs
 
+    # processing the onion sections:
+    for section in sections:
+        if not jm_single().config.has_option(section, "type") or \
+        not jm_single().config.get(section, "type").lower() == "onion":
+            continue
```

Comment:
If we go with my suggestion above, this can be replaced with just this (the comment can also be eliminated, since it is now self explanatory):

```suggestion
    for section in onion_sections:
```

Reply:
Comment as per https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/1182#r820208748